compiler/rustc_target/src/abi/call/mod.rs

```diff
@@ -517,11 +515,17 @@ impl<'a, Ty> ArgAbi<'a, Ty> {
         self.mode = Self::indirect_pass_mode(&self.layout);
     }
 
-    pub fn make_indirect_byval(&mut self) {
+    pub fn make_indirect_byval(&mut self, byval_align: Option<Align>) {
         self.make_indirect();
         match self.mode {
-            PassMode::Indirect { attrs: _, extra_attrs: _, ref mut on_stack } => {
+            PassMode::Indirect { ref mut attrs, extra_attrs: _, ref mut on_stack } => {
                 *on_stack = true;
+
+                // Some platforms, like 32-bit x86, change the alignment of the type when passing
+                // `byval`. Account for that.
+                if let Some(byval_align) = byval_align {
+                    attrs.pointee_align = Some(byval_align);
```

Comment:
Is it possible to assert that we are changing the alignment in a sound way? Like making sure it's only getting increased, not decreased or sth

Reply:
It actually can get decreased per the rules in https://github.com/rust-lang/rust/pull/103830/files#diff-b91fb3b3f9d0997cbceaf76264e8a6f0621fb178185f4e8a7c35e34d0f1a482cR61

I can add some kind of assertion though—perhaps that if we're on x86 it's always at least 4.

Reply:
Oh wow, that seems scary. 🤷 if that's what the ABI wants  let's do it.